.github/workflows/atm-check-queries-run.yml

```diff
@@ -1,13 +1,79 @@
 name: ATM Check Queries Run
 
-# This check is required, therefore we must run it on all PRs, even if only Markdown has changed.
+env:
+  AZURE_STORAGE_URL: "https://atmcodeqldata.blob.core.windows.net"
+  DB_NAME: "AmanSultanBaig/SignIn-SignUp-System-with-Nodejs"
+  DB_PATH: test_db
+  MODEL_BULDING_PACK_PATH: javascript/ql/experimental/adaptivethreatmodeling/modelbuilding
+  QUERY_SUITE: javascript/ql/experimental/adaptivethreatmodeling/src/codeql-suites/javascript-atm-code-scanning.qls
+
 on:
+  pull_request:
+    paths:
+      - "javascript/ql/experimental/adaptivethreatmodeling/modelbuilding/codeql-pack.lock.yml"
+      - "javascript/ql/experimental/adaptivethreatmodeling/modelbuilding/qlpack.yml"
+      - "javascript/experimental/adpativethreatmodeling/src/qlpack.yml"
+      - "javascript/experimental/adpativethreatmodeling/src/codeql-pack.lock.yml"
   workflow_dispatch:
 
 jobs:
-  hello-world:
+  run-atm-queries:
     runs-on: ubuntu-latest
 
     steps:
-      - name: foo
-        run: echo "Hello world"
+      - uses: actions/checkout@v3
+        with:
+          path: codeql-lib
+
+      - name: Install CodeQL CLI
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh extensions install github/gh-codeql
+          gh codeql download
+
+      - name: Download model pack
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          STORAGE_SAS: ${{ secrets.AZURE_BLOB_STORAGE_ATMCODEQLDATA_SAS }}
+        run: |
+          echo "::group::Download ATM model pack"
+          set -exu
+
+          # Get pack version and checksum
+          pack_version=$(yq '.dependencies.codeql/javascript-experimental-atm-model' ./codeql-lib/${MODEL_BULDING_PACK_PATH}/qlpack.yml )
+          model_checksum="${pack_version##*.}"
+          echo "Will use pack model ${pack_version} with model checksum ${model_checksum}."
+
+          # Download the model to the package cache
+          tmp_dir=$(mktemp -d)
+          gh codeql pack download codeql/javascript-experimental-atm-model@${pack_version}
```

Comment:
We can use the standard pack installation mechanism `gh codeql pack install javascript/ql/experimental/adaptivethreatmodeling/src` to do all of this :)

Reply:
I guess what I'm asking is: will it update our config file to trust the checksum?

Reply:
Using the standard interface for installing the model makes the workflow more robust (and is also more readable).  If the behind-the-scenes mechanism for installing the model changes, then the workflow doesn't break.  The workflow is currently not robust to the name of the model pack changing for instance, or the addition of a new pack dependency.

You're right that we'll still need to trust the checksum.  We should grab the checksum from `model.properties` instead of the version number though, since the version number of a released pack will not contain the checksum (it'll just be `0.4.0` for example).

Reply:
What is the recommended way to find the model.properties file? resolve using the query suite? or assume there's a single ml model?

Reply:
I'd recommend `gh codeql resolve extensions javascript/ql/experimental/adaptivethreatmodeling/src/NosqlInjectionATM.ql | jq -r '.models[0].checksum'`.  It's fine to make the assumption that there's a single ML model — that is unlikely to change and won't happen underneath us.

Reply:
I think we should avoid using a random query, it's more robust to change behind the scenes to use the suite.

Reply:
any idea what I'm missing here? https://github.com/github/codeql/actions/runs/3371044939/jobs/5592734461

Reply:
ugh it seems `actions/checkout` doesn't create a specific folder